lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -15,6 +15,11 @@
 #define ENABLE 1
 #define DISABLE 0
 
+/* Even though it's defined in minimal libc, the declaration of strtok_r appears to be missing in
+ * some cases, so we forward declare it here.
+ */
+extern char *strtok_r(char *str, const char *sep, char **state);
+
```

Comment:
Is this because some libraries need `_POSIX_C_SOURCE` defined to show `strtok_r()`.

Would it be better to define that macro before including `string.h`

Reply:
Hmm, so we'd need something like https://github.com/nrfconnect/sdk-nrf/blob/main/lib/date_time/date_time_core.c#L8 instead of forward declaration in all those places? Maybe that would indeed be cleaner than duplicating the declaration all around.

Reply:
That sure seems like a neater way of doing it, should I just do that in a separate commit for all the files where this is relevant?

Reply:
Okay, done!